src/platform/elements/value/Value.ts

```diff
@@ -23,25 +34,33 @@ export enum NOVO_VALUE_THEME { DEFAULT, MOBILE };
                 <i *ngFor="let icon of meta.icons" [class]="iconClass(icon)" (click)="onValueClick(icon)"></i>
             </div>
         </ng-container>
-    `
+    `,
 })
 export class NovoValueElement implements OnInit, OnChanges {
   @Input() data: any; // TODO use interface
   @Input() meta: any; // TODO use interface
   @Input() theme: NOVO_VALUE_THEME = NOVO_VALUE_THEME.DEFAULT;
+  @Input() retrieveText: Subject<any> = new Subject();
+  @Output() onRetrieveText: EventEmitter<string> = new EventEmitter<string>();
 
   type: NOVO_VALUE_TYPE;
   NOVO_VALUE_TYPE = NOVO_VALUE_TYPE;
   NOVO_VALUE_THEME = NOVO_VALUE_THEME;
   url: string;
   customClass: string = '';
 
+  constructor(public render: RenderPipe) {}
+
   ngOnInit() {
     if (Helpers.isEmpty(this.meta)) {
       this.meta = {
-        label: ''
+        label: '',
       };
     }
+
+    this.retrieveText.subscribe((event) => {
```

Comment:
What does this do exactly? Passing in subjects can hurt performance, especially since you are subscribing and not unsubscribing. What is the actual goal of this?